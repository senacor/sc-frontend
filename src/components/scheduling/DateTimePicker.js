import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import 'moment/locale/de';
import TextField from '@material-ui/core/TextField';

const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit
  }
});

class DateTimePicker extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      date: this.props.date,
      startTime: this.props.startTime,
      endTime: this.props.endTime
    };
  }

  onDateTimeChange = event => {
    let { name, value } = event.target;
    this.props.onDateTimeChange(name, value);
    this.setState({
      [name]: value
    });
  };

  render() {
    const { classes, intl } = this.props;

    return (
      <form>
        <TextField
          id="date"
          name="date"
          type="date"
          label={intl.formatMessage({
            id: 'datetimepicker.date'
          })}
          value={this.state.date}
          className={classes.textField}
          onChange={this.onDateTimeChange}
          InputLabelProps={{
            shrink: true
          }}
          required
          pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
        />
        <TextField
          type="time"
          name="startTime"
          className={classes.textField}
          id="appt-start-time"
          min="00:00"
          max="24:00"
          label={intl.formatMessage({
            id: 'datetimepicker.from'
          })}
          value={this.state.startTime}
          onChange={this.onDateTimeChange}
          required
        />
        <TextField
          type="time"
          name="endTime"
          className={classes.textField}
          id="appt-end-time"
          min="00:00"
          max="24:00"
          label={intl.formatMessage({
            id: 'datetimepicker.to'
          })}
          value={this.state.endTime}
          onChange={this.onDateTimeChange}
          required
        />
      </form>
    );
  }
}

export default withStyles(styles)(DateTimePicker);
