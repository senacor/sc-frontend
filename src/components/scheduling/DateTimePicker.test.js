import React from 'react';
import { shallow } from 'enzyme';
import DateTimePicker from './DateTimePicker';

describe('DateTimePicker', () => {
  it('should match snapshot', () => {
    let component = shallow(
      <DateTimePicker
        date={'2018-09-01'}
        startTime={'09:00'}
        endTime={'10:00'}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should call appointmentsSearch upon picking a date', () => {
    const setDateTimeMock = jest.fn();
    const component = shallow(
      <DateTimePicker onDateTimeChange={setDateTimeMock} />
    ).dive();

    component.find('#date').simulate('change', {
      target: { name: 'date', value: '2018-03-08' }
    });

    expect(setDateTimeMock.mock.calls.length).toBe(1);
    expect(setDateTimeMock.mock.calls[0][0]).toBe('date');
    expect(setDateTimeMock.mock.calls[0][1]).toBe('2018-03-08');
  });
});
