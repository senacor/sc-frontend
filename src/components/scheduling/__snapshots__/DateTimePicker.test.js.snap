// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker should match snapshot 1`] = `
<DateTimePicker
  classes={
    Object {
      "textField": "DateTimePicker-textField-1",
    }
  }
  date="2018-09-01"
  endTime="10:00"
  startTime="09:00"
/>
`;
